lua/ibl/init.lua

```diff
@@ -369,10 +389,67 @@ M.refresh = function(bufnr)
 
         last_whitespace_tbl = whitespace_tbl
 
-        -- #### make virtual text ####
-        local scope_start = row == scope_row_start
-        local scope_end = row == scope_row_end
-        if scope_start and scope then
+        --- current indent
+        if current_indent_enabled and row == lnum then
+            local current_indent_whitespace_tbl = vim.tbl_extend("keep", {}, whitespace_tbl)
+
+            while #current_indent_whitespace_tbl > 0 do
+                if indent.is_indent(current_indent_whitespace_tbl[#current_indent_whitespace_tbl]) then
+                    table.remove(current_indent_whitespace_tbl, #current_indent_whitespace_tbl)
+                    break
+                end
+                table.remove(current_indent_whitespace_tbl, #current_indent_whitespace_tbl)
+            end
+
+            current_indent_col_start_single = #current_indent_whitespace_tbl
+
+            for j = indent_state and #indent_state.stack or 0, 1, -1 do
+                if indent_state.stack[j].indent < current_indent_col_start_single then
+                    break
+                end
+                current_indent.start_row = indent_state.stack[j].row + 1
```

Comment:
```suggestion
                current_indent.start_row = indent_state.stack[j].row + 1
                if indent_state.stack[j].indent <= current_indent_col_start_single then
                    break
                end

```

This partially fixes the problem with multiple indents for me (I will make another note to fix the other problem).

Reply:
I ran into a problem with blanklines and made a quick fix (this hasn't really been tested yet).

Right now, I use
```lua
for j = indent_state and #indent_state.stack or 0, 1, -1 do
    current_indent.start_row = not blankline and indent_state.stack[j].row + 1 or current_indent.start_row
    if indent_state.stack[j].indent <= current_indent_col_start_single then
        break
    end
end
```
in that part of the code in my fork.